### ISSUE TITLE
Does not compile with Webpack 5 (using Webpack Encore)

### ISSUE THREAD

**Issue by jmsche** [froala/wysiwyg-editor#4156]
##### Expected behavior.
Webpack to successfully build Froala Editor

##### Actual behavior.
I'm getting this error:
> Module build failed: Module not found:
> ./node_modules/froala-editor/js/plugins.pkgd.min.js contains a reference to the file crypto.
> This file can not be found, please check it for typos or update it if the file got moved.

##### Steps to reproduce the problem.
Using this simple configuration: https://gist.github.com/jmsche/5ce6d06b4e48c7981bdf94e1fd05b2a9

I have the issue when upgrading Webpack Encore from 0.33.0 to 1.0.0+.
The main difference is that Encore uses Webpack 5 instead of Webpack <= 4.

##### Editor version.
Latest (3.2.6)

##### OS.
Linux

##### Browser.
N/A

##### More info
@Lyrkan helped me to fix it using Webpack configuration (so in my own files) but we think it should be fixed in Froala itself.
Here's something that may be related: https://webpack.js.org/blog/2020-10-10-webpack-5-release/#automatic-nodejs-polyfills-removed

Fixed it locally using these steps:
- adding `crypto-browserify readable-stream buffer` to my vendors
- adding this to the end of `webpack.config.js`:

```js
module.exports.resolve.fallback = {
    crypto: require.resolve('crypto-browserify'),
    stream: require.resolve('readable-stream')
};
```

**Comment by RicoFactset** [comment]
Hi,

I am using webpack encore 1.1.0 and get "TypeError: Cannot set property 'fallback' of undefined"

"resolve" is an unresolved variable in module.exports when using in webpack.config.js

module.exports.resolve.fallback = {
  crypto: require.resolve('crypto-browserify'),
  stream: require.resolve('readable-stream')
};

**Comment by RicoFactset** [comment]
I solved it with:

yarn add node-polyfill-webpack-plugin

const NodePolyfillPlugin = require("node-polyfill-webpack-plugin")

.addPlugin(new NodePolyfillPlugin())

**Comment by lpfrenette** [comment]
Thanks @RicoFactset , but Froala should correct this on their side. At the price they charge for this product.

**Comment by challet** [comment]
> Froala should correct this on their side. At the price they charge for this product.

Like many of the issues that sit in this repo

**Comment by mxlle** [comment]
This is an even bigger issue with Angular 12!
The webpack config is not exposed in Angular. 

I found a workaround, but it's 🥴 

I had to add a path to an empty file to the tsconfig:
```
"compilerOptions": {
    ...
    "paths": {
      "crypto": ["./empty"]
    }
},
```

**Comment by AkshayCM** [comment]
Thanks for your feedback.
This issue is not reproducible or have been found to be fixed on the latest Froala release v4.0.8. Can you please upgrade and check?

Use this for more info: https://gist.github.com/branflake2267/facdaf51ffa0026192e32e2c2804d239

**Comment by Panman82** [comment]
FWIW, the ember addon also had to deal with this issue.

TLDR, the existence of the [crypto file](https://github.com/froala/wysiwyg-editor/blob/v4.0.9/js/plugins/cryptojs.min.js) causes errors when Webpack tries to include all files in the plugins folder.

The [ember addon provides configuration options](https://github.com/froala/ember-froala-editor/blob/f48d4866e0941ad2566e599cb2f0e108e1cb29e9/README.md#configuration) to list the various plugins, languages, and themes that a consumer wants to use, and [the addon takes care of the actual import](https://github.com/froala/ember-froala-editor/blob/d0a4e1fb678087da263c0f8472047704ed2d1d65/addon/components/froala-editor.js#L13-L36). This is done using a [dynamic `import()` statement](https://github.com/emberjs/rfcs/blob/master/text/0507-embroider-v2-package-format.md#supported-subset-of-dynamic-import-syntax), which then causes Webpack to include all possible files in the build (including all files in the plugin folder). For whatever reason, `cryptojs.min.js` is not importable and causes an error. To work around this, you have to add [Webpack config to exclude that file](https://github.com/froala/ember-froala-editor/blob/f48d4866e0941ad2566e599cb2f0e108e1cb29e9/index.js#L28-L35).

```js
webpack: {
  resolve: {
    fallback: {
      crypto: false,
    },
  },
},
```

**Comment by AkshayCM** [comment]
Yes, we have verified with webpack 5 version and things were working well!

**Comment by richardkmichael** [comment]
See #4390 also.  Please fix this. :-)

**Comment by zahidkarim24** [comment]
We’re closing this issue due to inactivity. If you’re still experiencing this problem on the latest version of our editor, please feel free to reopen the issue and let us know. You can find details on the latest updates here: [Froala Editor Changelog](https://froala.com/wysiwyg-editor/changelog/).
